src/subdomains/generic/gs/gs.controller.ts

```diff
@@ -0,0 +1,29 @@
+import { Controller, Post, UseGuards, Body, Get, Query } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { ApiBearerAuth, ApiExcludeEndpoint } from '@nestjs/swagger';
+import { RoleGuard } from 'src/shared/auth/role.guard';
+import { UserRole } from 'src/shared/auth/user-role.enum';
+import { dbQueryDto } from './dto/db-query.dto';
+import { SupportReturnData } from './dto/support-return-data.dto';
+import { GsService } from './gs.service';
+
+@Controller('gs')
+export class GsController {
+  constructor(private readonly gsService: GsService) {}
+
+  @Post('db')
+  @ApiBearerAuth()
+  @ApiExcludeEndpoint()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.SUPPORT))
+  async getRawData(@Body() query: dbQueryDto): Promise<any> {
+    return await this.gsService.getRawData(query);
+  }
```

Comment:
I think I would have two separate routes for old and new logic, makes it much easier to read and refactor. WDYT?

Reply:
We can then change all existing to the new scheme and remove the old one.